### PR TITLE
Replacing map from set for duplicate check in row mappers

### PR DESCRIPTION
**Pull request by tanishi-egov**

### CODE REVIEW

**Review by Priyanka-eGov — APPROVED** [review]

File: health-services/census-service/src/main/java/digit/repository/rowmapper/CensusRowMapper.java [hunk 1]
```diff
@@ -82,14 +84,14 @@ public List<Census> extractData(ResultSet rs) throws SQLException, DataAccessExc
...
+     * @param additionalFieldSet A set to keep track of added AdditionalField objects.
      * @param censusEntry       The Census entry to which the AdditionalField object will be added.
      * @throws SQLException If an SQL error occurs.
      */
-    private void addAdditionalField(ResultSet rs, Map<String, AdditionalField> additionalFieldMap, Census censusEntry) throws SQLException {
+    private void addAdditionalField(ResultSet rs, Set<String> additionalFieldSet, Census censusEntry) throws SQLException {
         String additionalFieldId = rs.getString("additional_field_id");
 
-        if (ObjectUtils.isEmpty(additionalFieldId) || additionalFieldMap.containsKey(additionalFieldId)) {
+        if (ObjectUtils.isEmpty(additionalFieldId) || additionalFieldSet.contains(additionalFieldId)) {
```
Comment (Priyanka-eGov) [inline health-services/census-service/src/main/java/digit/repository/rowmapper/CensusRowMapper.java:94]: Add comments explaining this case

File: health-services/census-service/src/main/java/digit/repository/rowmapper/CensusRowMapper.java [hunk 2]
```diff
@@ -109,22 +111,22 @@ private void addAdditionalField(ResultSet rs, Map<String, AdditionalField> addit
...
+        additionalFieldSet.add(additionalFieldId);
     }
 
 
     /**
      * Adds a PopulationByDemographics object to the census entry based on the result set.
      *
      * @param rs                         The ResultSet containing the data.
-     * @param populationByDemographicMap A map to keep track of added PopulationByDemographics objects.
+     * @param populationByDemographicSet A set to keep track of added PopulationByDemographics objects.
```
Comment (Priyanka-eGov) [inline health-services/census-service/src/main/java/digit/repository/rowmapper/CensusRowMapper.java:122]: enhance this

File: health-services/plan-service/src/main/java/digit/repository/rowmapper/PlanConfigRowMapper.java [hunk 3]
```diff
@@ -29,10 +26,10 @@ public PlanConfigRowMapper(QueryUtil queryUtil) {
     @Override
     public List<PlanConfiguration> extractData(ResultSet rs) throws SQLException, DataAccessException {
         Map<String, PlanConfiguration> planConfigurationMap = new LinkedHashMap<>();
-        Map<String, File> fileMap = new LinkedHashMap<>();
-        Map<String, Operation> operationMap = new LinkedHashMap<>();
-        Map<String, Assumption> assumptionMap = new LinkedHashMap<>();
-        Map<String, ResourceMapping> resourceMappingMap = new LinkedHashMap<>();
+        Set<String> fileSet = new HashSet<>();
```
Comment (Priyanka-eGov) [inline health-services/plan-service/src/main/java/digit/repository/rowmapper/PlanConfigRowMapper.java:29]: add a comment here explaining why we need these sets

File: health-services/plan-service/src/main/java/digit/repository/rowmapper/PlanConfigRowMapper.java [hunk 4]
```diff
@@ -71,13 +72,13 @@ public List<PlanConfiguration> extractData(ResultSet rs) throws SQLException, Da
...
-     * @param fileMap         A map to keep track of added File objects.
+     * @param fileSet         A set to keep track of added File objects.
      * @throws SQLException If an SQL error occurs.
      */
-    private void addFiles(ResultSet rs, PlanConfiguration planConfigEntry, Map<String, File> fileMap) throws SQLException {
+    private void addFiles(ResultSet rs, PlanConfiguration planConfigEntry, Set<String> fileSet) throws SQLException {
         String fileId = rs.getString("plan_configuration_files_id");
 
-        if (ObjectUtils.isEmpty(fileId) || fileMap.containsKey(fileId)) {
+        if (ObjectUtils.isEmpty(fileId) || fileSet.contains(fileId)) {
```
Comment (Priyanka-eGov) [inline health-services/plan-service/src/main/java/digit/repository/rowmapper/PlanConfigRowMapper.java:81]: add comments for all these if cases

**Review by Priyanka-eGov — CHANGES_REQUESTED** [review]

**Review by shashwat-egov — APPROVED** [review]

**Pull request merged**